src/backend/distributed/utils/metadata_cache.c

```diff
@@ -1483,13 +1698,43 @@ ResetDistTableCacheEntry(DistTableCacheEntry *cacheEntry)
 
 	if (cacheEntry->shardIntervalArrayLength > 0)
 	{
```

Comment:
Maybe just return if `cacheEntry->shardIntervalArrayLength == 0`, since the lines below seem a bit cramped, indentation-wise.